File: 0704.二分查找/0704-二分查找.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        left = 0
        right = len(nums)-1
        while(left <= right):
            mid = (left+right)//2
            guess = nums[mid]
            if guess == target:
                return mid
            elif guess > target:
                right = mid-1
            elif guess < target:
                left = mid+1
        return -1

File: 0704.二分查找/0704-二分查找.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size()-1;  //×ó±ÕÓÒ±Õ
        while( left <= right ){
            int mid = left+(right-left)/2;  //·ÀÖ¹Òç³ö
            if( nums[mid] > target){
                right = mid-1;
            }
            else if( nums[mid] < target){
                left = mid + 1;
            }
            else{
                return mid;
            }
        }
        return -1;

    }
};